module-4/backend/week_two.md

```diff
@@ -5,18 +5,42 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's one difference between ES5 and ES6?
+  In ES5 you create variables using var and in ES6 you use let.
+
 2. What's the difference between asynchronous and synchronous JavaScript? 
+  Synchronous JavaScript runs in sequential back and forth order through a thread and asynchronous JavaScript runs concurrently because the browser provides more threads.
+
 3. What are the four pillars of Object Oriented programming?
+  Abstraction, encapsulation, inheritance, and polymorphism.
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+  You can use constructors to help you write object oriented code in javascript.
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
-6. What's a callback function and what are some reasons when we use/need callback functions?
+  Some key concepts to be aware o when refactoring JAvaScript is asynchronous issues and nested callbacks.
+
+ 6. What's a callback function and what are some reasons when we use/need callback functions?
+  A callback function is a function that is passed into another function as a parameter and called or executed inside the other function.
+
 7. What's the scope of variables in Javascript?
+  The scope of variables in Javascript are global and local.
+
 8. What's the difference between `==` and `===` in JavaScript?
+  == shows that the two objects are equal and === shows that the two objects are equal and the same type.
+
 9. Why do front end frameworks exist?
+  Front end frameworks exist to add structure to web applications with complex user interactions
 
 #### Review  
 
 10. Why do people say "HTTP is stateless"?
+  People say HTTP is stateless because the connection between the browser and server is lost once the transaction ends.
+
 11. Describe a RESTful API.
+  A restful api is an archetectural style used in web applications that uses HTTP requests to GET PUT POST and DELETE data.
+
 12. What are some main characteristics of a team following an agile workflow?
+  Some main characteristics of a team following an agile workflow are sprints, flexibility, and standups.
+
 13. What are some advantages **and** disadvantages to using OAuth to authenticate a user?
+  OAuth can be complicated to setup and relies on a third party service for authentication. However, it can streamline your authentication process and 'outsource' the security of it.
```

Comment:
yup - and "outsourcing" the security can be both an advantage and a disadvantage too!